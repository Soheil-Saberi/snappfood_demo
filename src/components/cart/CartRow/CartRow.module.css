.div {
  @apply mt-2 flex flex-row items-center justify-between;
}

.div p {
  @apply text-xs;
}
