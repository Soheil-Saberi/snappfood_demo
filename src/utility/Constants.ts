const Constants = {
  PACK_PRICE: 5000,
  TAX_PRICE: 12000,
  DELIVERY_PRICE: 15000,
  RESTAURANT_NAME: 'رستوران سالانی',
  RESTAURANT_STAR: '4.5',
  RESTAURANT_COMMENT: '500'
};

export default Constants;
