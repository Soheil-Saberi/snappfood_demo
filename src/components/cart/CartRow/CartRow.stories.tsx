import { ComponentMeta, ComponentStory } from '@storybook/react';
import CartRow from './CartRow';
import { string } from 'prop-types';

export default {
  component: CartRow,
  title: 'Cart/Row'
} as ComponentMeta<typeof CartRow>;

export const CartRowStory: ComponentStory<typeof CartRow> = (args) => {
  return (
    <div className="mt-4 gap-2 p-4 flex flex-col  border rounded-md shadow-md w-full h-fit">
      <CartRow title={args.title} price={args.price} />
    </div>
  );
};

CartRowStory.args = {
  title: 'هزینه دسته بندی',
  price: '500000'
};

CartRowStory.argTypes = {
  title: string.isRequired,
  price: string.isRequired
};
