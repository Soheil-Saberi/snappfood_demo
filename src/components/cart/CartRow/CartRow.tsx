import { FC, Fragment } from 'react';
// Style
import styles from './CartRow.module.css';
// Utils
import { camaPrice } from '../../../utility/Utils';

interface Props {
  title: string;
  price: string;
}

const CartRow: FC<Props> = ({ title, price }) => {
  return (
    <Fragment>
      {(price || price) && (
        <div className={styles.div}>
          <p>{title}</p>
          <p>{camaPrice(price)} تومان</p>
        </div>
      )}
    </Fragment>
  );
};

export default CartRow;
