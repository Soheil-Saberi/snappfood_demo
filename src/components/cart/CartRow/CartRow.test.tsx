import renderer from 'react-test-renderer';
import { describe, expect, test } from '@jest/globals';
import { cleanup } from '@testing-library/react';
// Component
import CartRow from './CartRow';
// Utils
import Constants from '../../../utility/Constants';

afterEach(() => {
  cleanup();
});

describe('CartRow Component', () => {
  test('Component Rendernig With Value Template', () => {
    const cartRow = renderer.create(<CartRow title="هزینه دسته بندی" price={Constants.PACK_PRICE.toString()} />);
    expect(cartRow.toJSON()).toMatchSnapshot();
  });

  test('Component Rendernig With Empty Value', () => {
    const cartRow = renderer.create(<CartRow title="" price="" />);
    expect(cartRow.toJSON()).toBeNull();
  });
});
