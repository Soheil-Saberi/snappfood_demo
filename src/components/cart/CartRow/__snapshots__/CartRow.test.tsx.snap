// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartRow Component Component Rendernig With Value Template 1`] = `
<div>
  <p>
    هزینه دسته بندی
  </p>
  <p>
    ۵,۰۰۰
     تومان
  </p>
</div>
`;
